### ISSUE TITLE
loss_radii = F.smooth_l1_loss(radii_pred[tcl_mask] / radii_map[tcl_mask], ones)   is inf

### ISSUE THREAD

**Issue by ustczhouyu** [princewang1994/TextSnake.pytorch#40]
I do not understand loss_radii = F.smooth_l1_loss(radii_pred[tcl_mask] / radii_map[tcl_mask], ones), what do radii_pred[tcl_mask] and radii_map[tcl_mask] mean? Can someone help me?

**Comment by WeihongM** [comment]
radii_pred[tcl_mask] means the predict radius map value in text center line region.
And  radii_map[tcl_mask] means the gt value.
I am not sure why the author use this method.
Is this loss function different from the below function?
`loss_radii = F.smooth_l1_loss(radii_pred[tcl_mask],  radii_map[tcl_mask])`
@princewang1994

**Comment by princevandev** [comment]
@WeihongM It is from the equation (8) in origin pape like:

![image](https://user-images.githubusercontent.com/15909795/77147132-98a1d400-6ac7-11ea-8886-7d0cd310d1cf.png)

You can see that in the first term of the `SmoothedL1`, it minimize (\hat{r} - r) / r，it can be transfered as  (\hat{r}/r -1). That is why I implement like this.

BTW, `loss_radii = F.smooth_l1_loss(radii_pred[tcl_mask], radii_map[tcl_mask])` only considers the absolute difference between `radii_pred` and `radii_map`, but (\hat{r} - r) / r considers the difference ratio of `r`. So it can be a way to reweight among text with different sizes. In other word, a huge text instance and a tiny text instance should be treated same in this loss.

Hope this can help you.

**Comment by WeihongM** [comment]
Thanks for your reply.
It is kind of you. I agree with your explanation!

**Comment by princevandev** [comment]
Well, I am closing this issue, thanks for your question! @WeihongM
